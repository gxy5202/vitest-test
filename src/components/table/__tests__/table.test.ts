import { expect, test, it } from 'vitest'
import { mount } from "@vue/test-utils";
import * as Vue from 'vue';
import Table from '../index.vue';

it('should work as expected', async () => {
    const wrapper = mount(Table, {
        props: { data: [{ name: 'a', age: '1' }] },
        template: `<el-table-column prop="name" label="名称"></el-table-column>`
    });

    // 渲染结果与快照做对比
    Vue.nextTick(() => {
        expect(wrapper.html()).toMatchSnapshot();
    });
})