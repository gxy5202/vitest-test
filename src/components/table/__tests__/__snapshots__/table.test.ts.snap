// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should work as expected 1`] = `
"<!-- <talbe class="cf-table">
        <tr v-for="item in data">{{ item }}</tr>
    </talbe> -->
<div class="el-table--fit el-table--enable-row-hover el-table--enable-row-transition el-table el-table--layout-fixed cf-table" data-prefix="el" role="cf-table">
  <div class="el-table__inner-wrapper">
    <div class="hidden-columns">{}</div>
    <div class="el-table__header-wrapper">
      <table class="el-table__header" border="0" cellpadding="0" cellspacing="0">
        <colgroup></colgroup>
        <thead class="">
          <tr class=""></tr>
        </thead>
      </table>
    </div>
    <div class="el-table__body-wrapper">
      <div class="el-scrollbar">
        <div class="el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
          <div class="el-scrollbar__view" style="display: inline-block; vertical-align: middle;">
            <table class="el-table__body" cellspacing="0" cellpadding="0" border="0" style="table-layout: fixed;">
              <colgroup></colgroup>
              <!--v-if-->
              <tbody tabindex="-1">
                <tr class="el-table__row"></tr>
              </tbody>
              <!--v-if-->
            </table>
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <transition-stub name="el-scrollbar-fade" appear="false" persisted="true" css="true">
          <div class="el-scrollbar__bar is-horizontal" style="display: none;">
            <div class="el-scrollbar__thumb" style="transform: translateX(0%);"></div>
          </div>
        </transition-stub>
        <transition-stub name="el-scrollbar-fade" appear="false" persisted="true" css="true">
          <div class="el-scrollbar__bar is-vertical" style="display: none;">
            <div class="el-scrollbar__thumb" style="transform: translateY(0%);"></div>
          </div>
        </transition-stub>
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="el-table__column-resize-proxy" style="display: none;"></div>
</div>"
`;
